### PR TITLE
Api simulate

### PR DESCRIPTION
**Pull request by enpitsuLin**
add `entries()`, `keys()`, `values()` and `forEach()`, also improve `toggle()` , [referenced from](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList)

Add test for them

### CODE REVIEW

**Comment by enpitsuLin** [comment]
Theres is an [`api`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/supports) does‘t implementated because I do not understand it already

And `add()` and `remove()`, which I haven't improved yet, should have more optional parameters

File: index.js [hunk 1]
```diff
@@ -58,6 +70,58 @@ function ClassList (node) {
     attribute = tokens.join(' ')
   }
 
+  function forEach (callback, thisArg) {
+    tokens.forEach(callback, thisArg)
+  }
+
+  function entries () {
+    return {
```
Comment (brechtcs) [inline index.js]: Just doing `return tokens.entries()` would be basically the same, no?

File: index.js [hunk 2]
```diff
@@ -58,6 +70,58 @@ function ClassList (node) {
...
+            }
+            return { done: true }
+          }
+        }
+      }
+    }
+  }
+
+  function keys () {
+    return {
```
Comment (brechtcs) [inline index.js]: Same as entries comment: `return tokens.keys()`

File: index.js [hunk 3]
```diff
@@ -58,6 +70,58 @@ function ClassList (node) {
...
+            }
+            return { done: true }
+          }
+        }
+      }
+    }
+  }
+
+  function values () {
+    return {
```
Comment (brechtcs) [inline index.js]: Same as entries and keys: `return tokens.values()`

**Comment by brechtcs** [comment]
Sorry for the delay. Added some minor remarks about the iterators, all the other additions look good to me.

File: index.js (hunk 2 above)
Comment (enpitsuLin) [reply index.js]: Oh, it seems to be so. I didn't notice that.

Comment (enpitsuLin) [reply index.js]: so it just only improve `toggle`🤣

**Pull request merged**